src/hotspot/cpu/x86/rdtsc_x86.cpp

```diff
@@ -105,7 +105,8 @@ static jlong initialize_frequency() {
     // for invariant tsc platforms, take the maximum qualified cpu frequency
     tsc_freq = (double)VM_Version_Ext::maximum_qualified_cpu_frequency();
     os_to_tsc_conv_factor = tsc_freq / os_freq;
-  } else {
+  }
+  if (tsc_freq == 0.0f) {
```

Comment:
tsc_freq is a double so use 0.0 not 0.0f

Probably worth a comment explaining this - something like:

// If no invariant TSC, or the system failed to report a useful maximum frequency, use an estimation.